src/WebJobs.Script.Analyzers/ReflectionAdapter/ReflectionHelpers.cs

```diff
@@ -0,0 +1,278 @@
+                var inner = arrayType.ElementType;
+
+                var innerRef = MakeFakeType(inner);
+
+                return innerRef.MakeArrayType();
+            }
+            if (p2.Kind == SymbolKind.NamedType)
+            {
+                string name = p2.Name;
+                string @namespace = GetFullMetadataName(p2.ContainingNamespace);
+
+                var type = p2 as INamedTypeSymbol;
+
+                if (type.IsGenericType)
+                {
+                    var typeArgs = new Type[type.TypeArguments.Length];
+                    for (int i = 0; i < typeArgs.Length; i++)
+                    {
+                        typeArgs[i] = MakeFakeType(type.TypeArguments[i]);
+                    }
+
+                    // Get Type Definition
+                    var asm = p2.ContainingAssembly;
+
+                    Type definition;
+                    Assembly asmRef;
+                    if (TryMapAssembly(asm, out asmRef))
+                    {
+                        // Reflection type. Important for unification with the binders
+                        var metadataName = type.GetFullMetadataName();
+                        definition = asmRef.GetType(metadataName);
+                    }
+                    else
+                    {
+                        // User generic type.
+                        definition = new FakeType(@namespace, p2.MetadataName, type);
+                    }
+
+                    return new GenericFakeType(definition, typeArgs);
+                }
+
+                Type result = new FakeType(@namespace, name, type);
+
+                return result;
+            }
+
+            throw new InvalidOperationException("Can't convert symbol type:" + p2.ToString());
+        }
+
+        private static bool IsRootNamespace(ISymbol s)
+        {
+            return s is INamespaceSymbol && ((INamespaceSymbol)s).IsGlobalNamespace;
+        }
+
+        // Get a Reflection adapter over the Roslyn symbol.
+        // This does not actually load the type via the CLR Loader.
+        // Return null on failure to create (likely a compiler error.)
+        public static Type GetParameterType(SyntaxNodeAnalysisContext context, ParameterSyntax parameterSyntax)
+        {
+            SymbolInfo parameterSymbol = context.SemanticModel.GetSymbolInfo(parameterSyntax.Type);
+            ISymbol p2 = parameterSymbol.Symbol;
+
+            if (p2 == null)
+            {
+                return null;
+            }
+
+            try
+            {
+                Type parameterType = MakeFakeType(p2); // throws if can't convert.
+
+                if (parameterSyntax.IsOutParameter())
+                {
+                    parameterType = parameterType.MakeByRefType();
+                }
+                return parameterType;
+            }
+            catch
+            {
+                return null; // Likely a compiler error
```

Comment:
thanks for adding this context - love it when i'm investigating something and i see explanations in the comments when visiting the code base. 😄

Reply:
lol in all fairness this one was Mike, but I'm 100% in agreement with you.